### PR TITLE
RUN-5069 Add window options-changed event

### PR DESCRIPTION
**Pull request by tomer-openfin**
#### Description of Change
**Added types, doc updates and a test for a new 'options-changed' event in window** 

The payload contains a diff array containing prev and new values for all changed options.

Related  [core PR](https://github.com/HadoukenIO/core/pull/754)

#### Checklist

- [x] PR description included
- [x] `npm test` passes
- [x] automated tests are [changed or added](https://testing-dashboard.openfin.co/#/app/dashboard)
- [x] relevant documentation was added
- [x] PR title starts with the JIRA ticket [pull request process](https://github.com/openfin/Internal-Wiki/wiki/Pull-Request-Process)
- [x] PR release notes describe the change in a way relevant to app-developers

#### Release Notes

Notes:  Types, doc updates and a test for a new 'options-changed' event

### CODE REVIEW

File: src/api/events/window.ts [hunk 1]
```diff
@@ -32,6 +32,16 @@ export interface WindowReloadedEvent<Topic, Type> extends WindowEvent<Topic, Typ
     url: string;
 }
 
+export interface WindowOptionsChanged<Topic, Type> extends WindowEvent<Topic, Type> {
+    diff: WindowOptionDiffObject[];
+}
+
+export interface WindowOptionDiffObject {
+    optionName: string;
```
Comment (pbaize) [inline src/api/events/window.ts]:
```suggestion
    optionName: keyof WindowOptions;
```

File: src/api/events/window.ts [hunk 2]
```diff
@@ -32,6 +32,16 @@ export interface WindowReloadedEvent<Topic, Type> extends WindowEvent<Topic, Typ
     url: string;
 }
 
+export interface WindowOptionsChanged<Topic, Type> extends WindowEvent<Topic, Type> {
+    diff: WindowOptionDiffObject[];
+}
+
+export interface WindowOptionDiffObject {
```
Comment (pbaize) [inline src/api/events/window.ts]: We could make WindowOptionDiffObject take a generic type which extends `keyof WindowOptions` and then make oldVal and newVal the type of that key on WindowOptions

File: test/window-event.test.ts [hunk 3]
```diff
@@ -3,47 +3,81 @@ import { conn } from './connect';
...
+            const expectedDiffObject = {
+                optionName,
+                oldVal: 1,
+                newVal: 0
+            };
+
+            return win.updateOptions(optionToUpdate)
+                .then(() => delayPromise())
+                .then(() => spy.args[0][0])
+                .then(payload => assert.deepEqual(payload.diff[0], expectedDiffObject));
```
Comment (datamadic) [inline test/window-event.test.ts]: Doe this need to be updated to the new `diff` as an object shape?

Comment (tomer-openfin) [reply test/window-event.test.ts]: yeah, will push a fix in a bit.

**Pull request merged**